### PR TITLE
Pullrequest

### PR DESCRIPTION
**Pull request by MKhlon**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+            <version>2.0.0</version>
```
Comment (oleksandr-fomenko) [inline pom.xml]: Optional comment: keep versions of the dependencies in the properties section

File: src/test/java/com/bookstore/utils/Utils.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.bookstore.utils;
+
+import com.bookstore.model.Product;
+import lombok.Data;
+
+
+@Data
```
Comment (oleksandr-fomenko) [inline src/test/java/com/bookstore/utils/Utils.java]: data annotation doesn't make any sense here

File: src/test/java/com/bookstore/utils/Utils.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class Utils {
```
Comment (oleksandr-fomenko) [inline src/test/java/com/bookstore/utils/Utils.java]: Better to rename it to TestDataUtils

Comment (oleksandr-fomenko) [reply src/test/java/com/bookstore/utils/Utils.java]: and add a private constructor + make class final

File: src/test/java/com/bookstore/utils/Messages.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.bookstore.utils;
+
+public class Messages {
```
Comment (oleksandr-fomenko) [inline src/test/java/com/bookstore/utils/Messages.java]: it's a good practice to add private constructors to such kind of classes and make class final

File: src/main/java/com/bookstore/model/Booking.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    @JoinColumn(name = "user_id")
+    private User user;
+
+    @Column(name = "delivery_address")
+    private String deliveryAddress;
+
+    @Column(name = "delivery_date")
+    private LocalDate date;
+
+    @JsonFormat(pattern = "HH:mm:ss.SSSSSSSSS")
```
Comment (oleksandr-fomenko) [inline src/main/java/com/bookstore/model/Booking.java]: In the classic MVC architecture data model != domain model. You need to have separate models for the DB layer and API layers and map them to each other.

File: src/main/java/com/bookstore/controller/BookingController.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+    public BookingController(BookingService bookingService) {
+        this.bookingService = bookingService;
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Booking> getBookingById(@PathVariable("id") Integer id) {
+        return this.bookingService.findById(id).map(b ->
+                        new ResponseEntity<>(b, HttpStatus.OK))
+                .orElse(new ResponseEntity<>(HttpStatus.NOT_FOUND));
```
Comment (oleksandr-fomenko) [inline src/main/java/com/bookstore/controller/BookingController.java:32]: FYI, not comment: it's a good practice to declare and throw custom exceptions for such cases and handle the using Spring ExceptionHandlers - https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc

**Review by oleksandr-fomenko — CHANGES_REQUESTED** [review]

**Comment by oleksandr-fomenko** [comment]
it's better to have a separate configuration application.yml for the application testing. Place it under src/test/resources, use h2DB instead of real DB

File: pom.xml (hunk 1 above)
Comment (MKhlon) [reply pom.xml]: done

File: src/test/java/com/bookstore/utils/Utils.java (hunk 2 above)
Comment (MKhlon) [reply src/test/java/com/bookstore/utils/Utils.java]: fixed

File: src/test/java/com/bookstore/utils/Utils.java (hunk 3 above)
Comment (MKhlon) [reply src/test/java/com/bookstore/utils/Utils.java]: fixed

File: src/test/java/com/bookstore/utils/Messages.java (hunk 4 above)
Comment (MKhlon) [reply src/test/java/com/bookstore/utils/Messages.java]: fixed

File: src/main/java/com/bookstore/model/Booking.java (hunk 5 above)
Comment (MKhlon) [reply src/main/java/com/bookstore/model/Booking.java]: fixed

File: src/main/java/com/bookstore/controller/BookingController.java (hunk 6 above)
Comment (MKhlon) [reply src/main/java/com/bookstore/controller/BookingController.java:32]: thanks, didn't change for now, but will know about it

**Review by oleksandr-fomenko — APPROVED** [review]

**Pull request merged**